prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -157,43 +188,27 @@ def generate_final_decision(
         )
         return output
 
-    def answer_binary_market(self, question: str) -> ProbabilityOutput:
-        outcomes = self.split_research_into_outcomes(question)
+    def answer_binary_market(self, question: str) -> ProbabilityOutput | None:
+        hypothetical_scenarios = self.get_hypohetical_scenarios(question)
+        conditional_scenarios = self.get_required_conditions(question)
+
+        sub_predictions = par_generator(
```

Comment:
I saw your comment regarding parallel execution.
I think this goes in the right direction, but would advise to use the standard parallelism from crewai (https://docs.crewai.com/core-concepts/Tasks/#asynchronous-execution) instead of managing futures ourselves.

Reply:
Keep in mind that their parallelism can be tricky - I had cases where just one task was resolved but it already called subsequential parts of the code.
So I would test it manually first and check which tasks can indeed be made parallel. Let me know if I can be of help.

Reply:
The only problem I see with current implementation is the high memory usage (having 1 agent per outcome, and if I'm not mistaken, 5-8 outcomes per question)?
With crew-ai each agent acts as Worker, hence it's less memory intensive. The downside is it takes longer, hence the tradeoff here RAM vs time of execution.

Reply:
Errors are fine. But should the other agent take any significant amount of memory? If they are working in parallel in both cases, they need to store the scraped web content, generations, etc. and otherwise, I hope there is not extra memory overload, let's try it out:

If I set `max_workers=1`, then:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/843abfb6-bf60-47df-bf09-6d0a2072fcb5)

And `max_workers=5`, then:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/44f12c84-95c2-47e5-8913-2a5b0224d6a9)

---

So I don't see any significant difference but memory profiling can be tricky, we will see if Kube goes OOM 🙈 

---

Btw the question was, `Will the sun rise tomorrow?`. Disappointing that there is 0.01% chance that it will not! 😄